### PR TITLE
feat: create product model and controller

### PR DESCRIPTION
**Pull request by mthanh2209**

### CODE REVIEW

File: javascript-practice/src/javascripts/models/product.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	}
+
+	/**
+	 * Method gets the initial product list
+	 * @param {Array} products
+	 */
+	initProducts(products) {
+		this.products = products;
+		this.notify(products);
+		return products;
```
Comment (hungnguyenhuu-agilityio) [inline javascript-practice/src/javascripts/models/product.js:13]: Before returning, should break the line

File: javascript-practice/src/javascripts/models/product.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class ProductModel {
```
Comment (hungnguyenhuu-agilityio) [inline javascript-practice/src/javascripts/models/product.js:1]: This is product model, but I can not see anything relate to the properties of product?

File: javascript-practice/src/javascripts/models/product.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import APIProduct from "../services/product";
+class ProductModel {
```
Comment (hungnguyenhuu-agilityio) [inline javascript-practice/src/javascripts/models/product.js:2]: Code styles

Comment (mthanh2209) [reply javascript-practice/src/javascripts/models/product.js:3]: I updated it

File: javascript-practice/src/javascripts/models/product.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+	constructor() {
+		this.apiService = new APIService("/products")
+	}
+
+	/**
+	 * Method to retrieve the list of product from the API.
+	 * @returns {Promise<Object[]>} A promise that resolves with the list of product.
+	 */
+	getProductList = () => this.apiService.getList()
+}
```
Comment (hungnguyenhuu-agilityio) [inline javascript-practice/src/javascripts/models/product.js:13]: The model but it does carry any structure of Product. Pls help to update. @mthanh2209

Comment (mthanh2209) [reply javascript-practice/src/javascripts/models/product.js:25]: Thank you. I updated it

File: javascript-practice/src/javascripts/models/product.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import APIService from "../services/service"
+
+export default class ProductModel {
+	constructor(id, title, price, description, categoryId, dimension, image) {
```
Comment (hungnguyenhuu-agilityio) [inline javascript-practice/src/javascripts/models/product.js:4]: So in you need to init all of params when you create this instance. Should not do that

Comment (hungnguyenhuu-agilityio) [reply javascript-practice/src/javascripts/models/product.js:4]: You missed that, @mthanh2209 ?

File: javascript-practice/src/javascripts/models/product.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import APIService from "../services/service"
+
+export default class ProductModel {
+	constructor(title, price) {
```
Comment (hungnguyenhuu-agilityio) [inline javascript-practice/src/javascripts/models/product.js:4]: Why we need to input the title and price?

File: javascript-practice/src/javascripts/models/product.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+export default class ProductModel {
+	constructor() {
+		this.id = id;
+		this.title = title;
+		this.price = price;
+		this.description = description;
+		this.categoryId = categoryId;
+		this.dimension = dimension;
+		this.image = image;
```
Comment (hungnguyenhuu-agilityio) [inline javascript-practice/src/javascripts/models/product.js:11]: Where are id, title, price, description, categoryId, dimentsion, image?

File: javascript-practice/src/javascripts/models/product.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import APIService from "../services/service";
+
+export default class ProductModel {
+	constructor(categoryId) {
```
Comment (hungnguyenhuu-agilityio) [inline javascript-practice/src/javascripts/models/product.js:4]: Why we need categoryId in every time we init the ProducModel instance?

**Pull request merged**